src/Utils.vala

```diff
@@ -34,4 +34,49 @@ namespace Display.Utils {
 
         return false;
     }
+
+    public static Gee.LinkedList<Display.MonitorMode> get_common_monitor_modes (Gee.LinkedList<Display.Monitor> monitors) {
+        var common_modes = new Gee.LinkedList<Display.MonitorMode> ();
+        double min_scale = get_min_compatible_scale (monitors);
+        bool first_monitor = true;
+        foreach (var monitor in monitors) {
+            if (first_monitor) {
+                foreach (var mode in monitor.modes) {
+                    if (min_scale in mode.supported_scales) {
+                        common_modes.add (mode);
+                    }
+                }
+
+                first_monitor = false;
+            } else {
+                var to_remove = new Gee.LinkedList<Display.MonitorMode> ();
+                foreach (var mode_to_check in common_modes) {
+                    bool mode_found = false;
+                    foreach (var monitor_mode in monitor.modes) {
+                        if (mode_to_check.width == monitor_mode.width &&
+                            mode_to_check.height == monitor_mode.height) {
+                            mode_found = true;
```

Comment:
`break` here. There's no need to search for more modes if there's already one matching.

Reply:
thanks, indeed. I'm always afraid of using `break` inside nested loops, I miss the `break n` syntax